crates/core/tedge_agent/src/file_transfer_server/error.rs

```diff
@@ -24,8 +23,67 @@ pub enum FileTransferError {
     BindingAddressInUse { address: std::net::SocketAddr },
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum FileTransferRequestError {
+    #[error(transparent)]
+    FromIo(#[from] std::io::Error),
+
+    #[error("Cannot delete: {path:?} is a directory, not a file")]
+    CannotDeleteDirectory { path: RequestPath },
+
+    #[error("Cannot upload: {path:?} is a directory, not a file")]
+    CannotUploadDirectory { path: RequestPath },
+
+    #[error("Request to delete {path:?} failed: {source}")]
+    OtherDelete {
+        source: std::io::Error,
+        path: RequestPath,
+    },
+
+    #[error("Request to upload to {path:?} failed: {source:?}")]
+    OtherUpload {
+        source: anyhow::Error,
+        path: RequestPath,
+    },
```

Comment:
suggestion: I'd be for keeping old just `Delete` or `Upload` for these variants, I think it's fine if they're just anyhow errors. Current names kinda make one think _"`OtherUpload`? Well, what about the first upload?"_

Reply:
Sure, I won't change this right now as I don't want to delay merging further, but I'll make the change when I do the next step to add TLS.